File: apps/android/core/models/order.kt
package com.shoplift.app.core.models

import kotlinx.serialization.Serializable

@Serializable
enum class OrderStatus(val displayName: String) {
    PENDING("Pending"),
    CONFIRMED("Confirmed"),
    PREPARING("Preparing"),
    READY_FOR_PICKUP("Ready for Pickup"),
    OUT_FOR_DELIVERY("Out for Delivery"),
    DELIVERED("Delivered"),
    CANCELLED("Cancelled")
}

@Serializable
data class Order(
    val id: String,
    val customerId: String,
    val restaurantId: String,
    val items: List<OrderItem>,
    val status: OrderStatus,
    val paymentMethod: String,
    val paymentGateway: String? = null,
    val paymentIntentId: String? = null,
    val subtotal: Int,
    val deliveryFee: Int,
    val discount: Int,
    val total: Int,
    val promoCode: String? = null,
    val deliveryAddress: Address,
    val estimatedDeliveryTime: String? = null,
    val notes: String? = null,
    val createdAt: String,
    val confirmedAt: String? = null,
    val deliveredAt: String? = null
) {
    val totalDisplay: String
        get() = "$${String.format("%.2f", total / 100.0)}"
}

@Serializable
data class OrderItem(
    val itemId: String,
    val name: BilingualString,
    val quantity: Int,
    val unitPrice: Int,
    val selectedModifiers: List<SelectedModifier>? = null,
    val specialInstructions: String? = null,
    val subtotal: Int
) {
    val displayName: String
        get() = name.en
    
    val subtotalDisplay: String
        get() = "$${String.format("%.2f", subtotal / 100.0)}"
}

@Serializable
data class SelectedModifier(
    val groupName: String,
    val optionName: String,
    val priceAdjustment: Int
)

@Serializable
data class Address(
    val street: String,
    val district: String,
    val city: String,
    val coordinates: Coordinates? = null
)

@Serializable
data class Coordinates(
    val lat: Double,
    val lng: Double
)

File: apps/ios/core/models/order.swift
import Foundation

enum OrderStatus: String, Codable, CaseIterable {
    case pending = "PENDING"
    case confirmed = "CONFIRMED"
    case preparing = "PREPARING"
    case readyForPickup = "READY_FOR_PICKUP"
    case outForDelivery = "OUT_FOR_DELIVERY"
    case delivered = "DELIVERED"
    case cancelled = "CANCELLED"
}

struct Order: Codable, Identifiable {
    let id: String
    let customerId: String
    let restaurantId: String
    let items: [OrderItem]
    let status: OrderStatus
    let paymentMethod: String
    let paymentGateway: String?
    let paymentIntentId: String?
    let subtotal: Int
    let deliveryFee: Int
    let discount: Int
    let total: Int
    let promoCode: String?
    let deliveryAddress: Address
    let estimatedDeliveryTime: String?
    let notes: String?
    let createdAt: String
    let confirmedAt: String?
    let deliveredAt: String?
    
    // Computed properties for UI
    var totalDisplay: String {
        return String(format: "%.2f", Double(total) / 100.0)
    }
    
    var statusDisplay: String {
        switch status {
        case .pending: return "Pending"
        case .confirmed: return "Confirmed"
        case .preparing: return "Preparing"
        case .readyForPickup: return "Ready for Pickup"
        case .outForDelivery: return "Out for Delivery"
        case .delivered: return "Delivered"
        case .cancelled: return "Cancelled"
        }
    }
}

struct OrderItem: Codable {
    let itemId: String
    let name: BilingualString
    let quantity: Int
    let unitPrice: Int
    let selectedModifiers: [SelectedModifier]?
    let specialInstructions: String?
    let subtotal: Int
    
    var displayName: String {
        return name.en
    }
    
    var subtotalDisplay: String {
        return String(format: "%.2f", Double(subtotal) / 100.0)
    }
}

struct SelectedModifier: Codable {
    let groupName: String
    let optionName: String
    let priceAdjustment: Int
}

struct Address: Codable {
    let street: String
    let district: String
    let city: String
    let coordinates: Coordinates?
}

struct Coordinates: Codable {
    let lat: Double
    let lng: Double
}
